packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx

```diff
@@ -53,7 +57,7 @@ export function TableNG(props: TableNGProps) {
   const columnWidth = useMemo(() => {
     setRevId(revId + 1);
     return fieldConfig?.defaults?.custom?.width || 'auto';
-  }, [fieldConfig]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [fieldConfig, props.data]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
This is going to trigger a react-data-grid re-render with every data update. Is there an alternative approach? Perhaps check for a column width change before incrementing the revId?

Reply:
I noticed that without it `<DataGrid />` didn't update (same key - nothing to update) when I added a column:

https://github.com/user-attachments/assets/f898e67d-d1b8-4f0f-a77a-ba4dbad72949

And I don't see harm to trigger re-render when data changed 😁

Reply:
It is interesting that when I update existing data - it works:

https://github.com/user-attachments/assets/a707511d-045e-4eca-9520-0008f1c7a832

Reply:
Your latest commit works great here!